src/test/java/com/cag/cagbackendapi/services/user/UserServiceTest.kt

```diff
@@ -18,37 +18,43 @@ import org.junit.jupiter.api.assertThrows
 import org.junit.jupiter.api.extension.ExtendWith
 import org.mockito.InjectMocks
 import org.mockito.junit.jupiter.MockitoExtension
+import org.springframework.security.crypto.password.PasswordEncoder
 import java.util.*
 
 @ExtendWith(MockitoExtension::class)
 class UserServiceTest {
 
     private var userDao: UserDao = mock()
+    private var passwordEncoder: PasswordEncoder = mock()
 
     @InjectMocks
     private lateinit var userService: UserService
 
     @Test
     fun registerUser_validUser_logsAndSucceeds() {
         // assemble
-        val inputUser = UserRegistrationDto("testy", "tester", "testytester@aol.com", true)
+        val inputPass = "password"
+        val encodedPass = "encodedPass"
+        val inputUser = UserRegistrationDto("testy", "tester", "testytester@aol.com", inputPass, true)
         val resultUser = UserDto(UUID.randomUUID(), "testy", "tester", "testytester@aol.com", true, null, null, true)
 
+        whenever(passwordEncoder.encode(inputPass)).thenReturn(encodedPass)
         whenever(userDao.saveUser(inputUser)).thenReturn(resultUser)
 
         // act
         userService.registerUser(inputUser)
 
         // assert
+        verify(passwordEncoder).encode(inputPass)
         verify(userDao).saveUser(inputUser)
         verifyNoMoreInteractions(userDao)
     }
 
     @Test
-    fun registerUser_missingFirstNameAndLastNameAndEmailAndAgreed18_BadRequest() {
+    fun registerUser_missingFirstNameAndLastNameAndEmailAndPasswordAndAgreed18_BadRequest() {
```

Comment:
Tested if password `null` but not the `userRegistrationDto.getPass().isBlank()` condition.

Reply:
Added a test! Good catch